tests/isis_integration_test.go

```diff
@@ -59,4 +121,199 @@ func TestServer(t *testing.T) {
 	}, dst) {
 		return
 	}
+
+	// send a hello from neighborA
+	eth0.SendFromRemote(neighborA.mac, []byte{
+		0x00, // DSAP
+		0x00, // CSAP
+		0x00, // CF
+		// ISO 10589 header
+		131, // Intradomain Routing Protocol Discriminator: ISIS
+		20,  // Length indicator
+		1,   // Version / Protocol ID Extension
+		0,   // ID Length
+		17,  // Type
+		1,   // Version
+		0,   // Reserved
+		0,   // Maximum Area Addresses
+		// ISIS hello
+		2,                          // Level 2 only
+		222, 173, 190, 239, 255, 1, // System ID
+		0, 16, // Holding timer
+		0, 42, // PDU length
+		2, // Local Circuit ID
+		// P2P Adj. State TLV
+		240, // Type
+		5,   // Length
+		2,   // Adj State down
+		0, 0, 0, 0,
+		// Protocols supported TLV
+		129, // Type
+		2,   // Length
+		204, // IPv4
+		142, // IPv6
+		// IP Interface addresses TLV
+		132,              // Type
+		4,                // Length
+		169, 254, 100, 1, // IP Address
+		// Area Addresses TLV
+		1,       // Type
+		3,       // Length
+		2,       // Area length
+		0x49, 0, // Area
+	})
+
+	// checking if the adjancency exists
+	time.Sleep(time.Second)
+	for _, a := range s.GetAdjacencies() {
+		assert.Equal(t, neighborA.mac.String(), a.Address.String())
+		assert.Equal(t, "eth0", a.InterfaceName)
+		assert.Equal(t, neighborA.systemID.String(), a.SystemID.String())
+		assert.Equal(t, packet.P2PAdjStateInit, int(a.Status))
+	}
+
+	// let's see if the neighborA is not listed in the hello packet
+	dst, pkt = eth0.ReceiveAtRemote()
+	if !assert.Equal(t, []byte{
+		// ISO 10589 header
+		131, // Intradomain Routing Protocol Discriminator: ISIS
+		20,  // Length indicator
+		1,   // Version / Protocol ID Extension
+		0,   // ID Length
+		17,  // Type
+		1,   // Version
+		0,   // Reserved
+		0,   // Maximum Area Addresses
+		// ISIS hello
+		2,                      // Level 2 only
+		12, 12, 12, 13, 13, 13, // System ID
+		0, 16, // Holding timer
+		0, 52, // PDU length
+		1, // Local Circuit ID
+		// P2P Adj. State TLV <--- Important part
+		240,        // Type
+		15,         // Length
+		1,          // Adj State down
+		0, 0, 0, 0, // extended local circuit id
+		222, 173, 190, 239, 255, 1, // Neighbor system ID
+		0, 0, 0, 0, // Neighbor extended local circuit id
+		// Protocols supported TLV
+		129, // Type
+		2,   // Length
+		204, // IPv4
+		142, // IPv6
+		// IP Interface addresses TLV
+		132,              // Type
+		4,                // Length
+		169, 254, 100, 0, // IP Address
+		// Area Addresses TLV
+		1,       // Type
+		3,       // Length
+		2,       // Area length
+		0x49, 0, // Area
+	}, pkt) {
+		return
```

Comment:
Why just `return` and not erroring out?

Reply:
assert.Equal logs what is wrong where. Thus just exiting here.
Checking the following assertions doesn't make sense here if the previous failed because they depend on the previous ones. if assertion A fails it implies all the following will most probably fail and the outputs of those failed asserts would just pollute the tests output.